### ISSUE TITLE
Replace chosen.js with select2

### ISSUE THREAD

**Issue by humancompanion** [USAJOBS/design-system#128]
chosen.js has a long thread on the lack of accessibility (https://github.com/harvesthq/chosen/issues/264) and multiple other issues, a few with pull requests, but the maintainers have not adequately addressed the problems. Thus I'm going to replace it with select2: https://select2.org/

**Comment by humancompanion** [comment]
This is mostly done but, I still need to update the button-select variation which is used in ATP: http://usajobs.github.io/design-system/buttons/#button-select

**Comment by scottblinch** [comment]
Hey, we stumbled upon this while searching for accessibility issues with select2. We found that select2, in addition to chosen, isn't exactly great for screen readers. We're still doing research and testing on this, but selectWoo (a select2 fork by WooCommerce) seems promising so far: https://github.com/woocommerce/selectwoo

Thought you'd like to know!

**Comment by humancompanion** [comment]
* [ ] Fix the icon in the select menus to match the Standards.
